src/etcd/client.py

```diff
@@ -757,7 +757,8 @@ def eternal_watch(self, key, index=None, recursive=None):
         local_index = index
         while True:
             response = self.watch(key, index=local_index, timeout=0, recursive=recursive)
-            local_index = response.modifiedIndex + 1
+            if index:
```

Comment:
If not set local_index means that the subsequent should start watching for events starting at the current index, whatever that may be.
If set local_index means than the subsequent will wait this index, etcd only keeps the responses of the most recent 1000 events across all etcd keys.If we miss all the 1000 events will cause error.